homeassistant/components/songpal/media_player.py

```diff
@@ -133,6 +138,28 @@ async def async_will_remove_from_hass(self) -> None:
         """Run when entity will be removed from hass."""
         await self._dev.stop_listen_notifications()
 
+    async def _get_sound_modes_info(self):
+        """Get available sound modes and the active one."""
+        settings = await self._dev.get_sound_settings("soundField")
+        if isinstance(settings, Setting):
+            settings = [settings]
+
+        sound_modes = {}
+        active_sound_mode = None
+        for setting in settings:
+            cur = setting.currentValue
+            for opt in setting.candidate:
+                if opt.value == cur:
+                    active_sound_mode = opt.value
+                if not opt.isAvailable:
+                    continue
```

Comment:
It's been a while since I have touched the library code, but should `isAvailable` check be done prior to the conditional above assigning `active_sound_mode`?

Reply:
I just tested with my soundbar, and yes if an option was selected (Cinema) and I change the input to bluetooth receiver that doesn't support Cinema, the soundfield value is not Cinema anymore, and it became Cinema again if I change again to an input that support it. So you are right, better to do the check before.